Assignment/Assignment.Tests/PingProcessTests.cs

```diff
@@ -59,42 +60,59 @@ public void RunTaskAsync_Success()
     {
         // Do NOT use async/await in this test.
         // Test Sut.RunTaskAsync("localhost");
+        Task<PingResult> result = Sut.RunTaskAsync("localhost");
+        AssertValidPingOutput(result.Result);
     }
 
     [TestMethod]
     public void RunAsync_UsingTaskReturn_Success()
     {
         // Do NOT use async/await in this test.
-        PingResult result = default;
         // Test Sut.RunAsync("localhost");
-        AssertValidPingOutput(result);
+        Task<PingResult> result = Sut.RunAsync("localhost");
+        AssertValidPingOutput(result.Result);
     }
 
     [TestMethod]
-#pragma warning disable CS1998 // Remove this
     async public Task RunAsync_UsingTpl_Success()
     {
-        // DO use async/await in this test.
-        PingResult result = default;
+        
+        PingResult result = await Sut.RunAsync("localhost");
 
-        // Test Sut.RunAsync("localhost");
+       
         AssertValidPingOutput(result);
     }
-#pragma warning restore CS1998 // Remove this
 
 
     [TestMethod]
     [ExpectedException(typeof(AggregateException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
     {
-        
+        CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
+        cancellationTokenSource.Cancel(); 
+        Task<PingResult> result = Sut.RunAsync("localhost",  cancellationToken);
+        AssertValidPingOutput(result.Result);

```

Comment:
To properly test Cancel you should probably put it after the call to Sut.RunAsync. If you do that though your test is not going to pass, so you need to modify a couple things. It also seems unnecessary to me that you call AssertVaidPingOutput here. You will also want to wait (or .Result) on the Task to make sure that the method will either complete or will throw the exception. 
```suggestion
        CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
        CancellationToken cancellationToken = cancellationTokenSource.Token;
        Task<PingResult> result = Task.Run(() => Sut.RunAsync("localhost", cancellationToken));
        cancellationTokenSource.Cancel();
        result.Wait();
```